cmake/flags.cmake

```diff
@@ -257,6 +257,23 @@ function(setupBuildFlags)
       target_link_options(c_settings INTERFACE
         -fsanitize=address
       )
+    elseif(OSQUERY_ENABLE_THREAD_SANITIZER)
+      target_compile_options(cxx_settings INTERFACE
+        -fsanitize=thread
+        -g
```

Comment:
I wouldn't add this, `-g` is handled by setting `CMAKE_BUILD_TYPE=RelWithDebInfo` or `CMAKE_BUILD_TYPE=Debug`.